### PR TITLE
Change config dir for snap

### PR DESCRIPTION
**Pull request by bartoszWojciechO**

### CODE REVIEW

File: internal/filesystem.go [hunk 1]
```diff
@@ -217,16 +217,38 @@ func cleanPidFile(pidFile string) {
 	}
 }
 
+// EnsureDir creates all directories along the path excluding the last element.
 func EnsureDir(path string) error {
 	dir, err := filepath.Abs(filepath.Dir(path))
```
Comment (mariusSincovici) [inline internal/filesystem.go:222]: Wouldn't be easier to just call `EnsureDirFull` and then `ensureDir` can be combined with it, e.g.
```
func EnsureDir(path string) error {
   return EnsureDirFull(filepath.Dir(path))
}
```

File: internal/filesystem.go [hunk 2]
```diff
@@ -217,16 +217,38 @@ func cleanPidFile(pidFile string) {
...
-	if _, err := os.Stat(dir); os.IsNotExist(err) {
-		err = os.MkdirAll(dir, PermUserRWX)
+	if err := ensureDir(dir); err != nil {
+		return fmt.Errorf("ensuring dir: %w", err)
+	}
+	return nil
+}
+
+func ensureDir(path string) error {
+	if _, err := os.Stat(path); os.IsNotExist(err) {
```
Comment (mariusSincovici) [inline internal/filesystem.go:245]: do you know why is `os.Stat` needed?
`os.MkdirAll` does internally a `Stat` and if the path exists and it is a folder will return no error.
But in our case if at the path something exists, including a file, then the folders are not created anymore.

File: internal/filesystem.go (hunk 1 above)
Comment (Savolro) [reply internal/filesystem.go:222]: Is there any case where we would not want to use EnsureDirFull?

Comment (bartoszWojciechO) [reply internal/filesystem.go]: There are a few cases where a path with a file at the end is given.

**Review by Savolro — APPROVED** [review]

**Review by keliramu — APPROVED** [review]

**Review by mariusSincovici — APPROVED** [review]

**Pull request merged**